docs/tutorials/03_shaders/viz_sdf_cylinder.py

```diff
@@ -0,0 +1,288 @@
+# We split the shader code into 4 different variables which correspond to
+# vertex and fragment shader declaration and implementation.
+#
+# Vertex shaders perform basic processing of each individual vertex.
+
+sdf_cylinder_vert_dec = \
+    '''
+    
+    /* SDF vertex shader declaration */
+    
+    //VTK::ValuePass::Dec
+    
+    in vec3 center;
+    in vec3 direction;
+    
+    out vec4 vertexMCVSOutput;
+    out vec3 centerWCVSOutput;
+    out vec3 directionVSOutput;
+    
+    '''
+
+sdf_cylinder_vert_impl = \
+    '''
+    
+    /* SDF vertex shader implementation */
+    
+    //VTK::ValuePass::Impl
+    
+    vertexMCVSOutput = vertexMC;
+    centerWCVSOutput = center;
+    directionVSOutput = direction;
+    
+    '''
+
+###############################################################################
+# Fragment shaders run on each of the pixels that the object occupies on the
+# screen.  For each sample of the pixels covered by a primitive, a ‘fragment’
+# is generated.
+
+sdf_cylinder_frag_dec = \
+    '''
+    /* SDF fragment shader declaration */
+    
+    //VTK::ValuePass::Dec
+    in vec4 vertexMCVSOutput;
+    
+    in vec3 centerWCVSOutput;
+    in vec3 directionVSOutput;
+    
+    uniform mat4 MCVCMatrix;
+    
+    
+    // A rotation matrix is used to transform our position vectors
+    // around a given 3D axis
+    mat4 rotationAxisAngle( vec3 v, float angle )
+    {
+        float s = sin(angle);
+        float c = cos(angle);
+        float ic = 1.0 - c;
+    
+        return mat4( v.x*v.x*ic + c,     v.y*v.x*ic - s*v.z, v.z*v.x*ic + s*v.y, 0.0,
+                     v.x*v.y*ic + s*v.z, v.y*v.y*ic + c,     v.z*v.y*ic - s*v.x, 0.0,
+                     v.x*v.z*ic - s*v.y, v.y*v.z*ic + s*v.x, v.z*v.z*ic + c,     0.0,
+                     0.0,                0.0,                0.0,                1.0 );
+    }
+    
+    
+    // SDF for the cylinder
+    float sdCylinder( vec3 p, float h, float r )
+    {
+        vec2 d = abs(vec2(length(p.xz),p.y)) - vec2(h,r);
+        return min(max(d.x,d.y),0.0) + length(max(d,0.0));
+    }
+    
+    
+    // This is used on calculations for surface normals of the cylinder
+    float map( in vec3 position )
+    {
+        mat4 rot = rotationAxisAngle( normalize(directionVSOutput), 90);
+    
+        // this allows us to accommodate more than one object in the world space
+        vec3 pos = (rot*vec4(position - centerWCVSOutput, 0.0)).xyz;
+    
+        // distance to the cylinder
+        return sdCylinder(pos, 0.6, 0.8);
+    }
+    
+    
+    // We need surface normals when doing lighting of the scene
+    vec3 calculateNormal( in vec3 position )
+    {
+        vec2 e = vec2(0.001, 0.0);
+        return normalize( vec3( map(position + e.xyy) - map(position - e.xyy),
+                                map(position + e.yxy) - map(position - e.yxy),
+                                map(position + e.yyx) - map(position - e.yyx)));
+    }
```

Comment:
This is the same central differences method used to calculate normals. Use `fury/shaders/sdf/central_diffs.frag` as in #587.